src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_cpu.cpp

```diff
@@ -181,13 +169,7 @@ size_t BrgemmCPU::get_offset_scratch() const {
     return get_input_offset(2);
 }
 
-BrgemmCPU::ShapeInfer::ShapeInfer(const std::shared_ptr<ov::Node>& n) : Brgemm::ShapeInfer(n) {
-    const auto& brg = ov::as_type_ptr<BrgemmCPU>(n);
-    OPENVINO_ASSERT(brg, "Got invalid node in BrgemmCPU::ShapeInfer");
-    const auto brgemm_copy = brg->is_with_data_repacking() ? brg->get_brgemm_copy() : nullptr;
-    if (brgemm_copy)
-        m_io_layouts[1] = snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(brgemm_copy->input(0))->get_layout();
-}
+BrgemmCPU::ShapeInfer::ShapeInfer(const std::shared_ptr<ov::Node>& n) : Brgemm::ShapeInfer(n) {}
```

Comment:
If shapeInfer is declared inside a node, it means that it is unique for this particular node. If the same shapeInfer is shared between several nodes, then it must be implemented separately in shape_infer_instances, and used in the factory's mapping for both of the nodes. This would help to keep shapeInfer pipeline concise and transparent, so let's try to keep it this way.
Note that we can split shapeInfer implementations into several files if necessary.

Reply:
Done, thanks